lib/matplotlib/axes/_axes.py

```diff
@@ -3723,9 +3723,10 @@ def boxplot(self, x, notch=None, sym=None, vert=None, whis=None,
         Draw a box and whisker plot.
 
         The box extends from the first quartile (Q1) to the third
-        quartile (Q3) of the data, with a line at the median.  The
-        whiskers extend from the box by 1.5x the inter-quartile range
-        (IQR).  Flier points are those past the end of the whiskers.
+        quartile (Q3) of the data, with a line at the median. The
+        whiskers extend from the box to the farthest point within
+        1.5x the inter-quartile range (IQR) distance. Flier points
+        are those past the end of the whiskers.
```

Comment:
```suggestion
        quartile (Q3) of the data, with a line at the median. The
        whiskers extend from the box to 1.5x the inter-quartile 
        range (IQR) distance from the box. Flier points
        are those past the end of the whiskers.
```

Reply:
also, is it the farthest data point and hence why you said "farthest point"?

Reply:
Yes. 
For example if all points fit inside the [Q1 - Q3] range there would be no whiskers at all. An example of such data could be `vals = [1,1,1,3,3,3]`

In that sense the ASCII art is correct in spirit but does not address the weird cases.

Reply:
So maybe just specify "data point" and distance from edge of box?

Reply:
So:

The whiskers extend from the box to the farthest data point 
lying within 1.5x the inter-quartile range (IQR) from the box

?

Reply:
Yes, that would make sense to me...

Reply:
There is a commit ready with the wording above